balancer/grpclb/grpclb_test.go

```diff
@@ -1241,10 +1243,11 @@ func testGRPCLBEmptyServerList(t *testing.T, svcfg string) {
 		grpc.WithTransportCredentials(&serverNameCheckCreds{}),
 		grpc.WithContextDialer(fakeNameDialer),
 	}
-	cc, err := grpc.DialContext(ctx, r.Scheme()+":///"+beServerName, dopts...)
+	cc, err := grpc.NewClient(r.Scheme()+":///"+beServerName, dopts...)
 	if err != nil {
-		t.Fatalf("Failed to dial to the backend %v", err)
+		t.Fatalf("Failed to create a newclient to the backend %v", err)
 	}
+	cc.Connect()
```

Comment:
Same here, we can replace the next `r.UpdateState` and avoid explicitly connecting here. Please check if we can avoid using `cc.Connect` in other places too.

Reply:
Fixed this.